### PR TITLE
*: supports build pd-server with fail point

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: Close #4399

### What is changed and how does it work?

```commit-message
In order can use fail points in testing, need to build a special binary
- supports build pd-server with fail point
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)

```
make pd-server-failpoint 
➜  bin git:(support-make-failpoint) tiup playground v7.1.1  --tiflash=0  --pd=1 --pd.binpath=./pd-server
➜  bin git:(support-make-failpoint) curl http://127.0.0.1:2379/pd/api/v1/fail
malformed request URI
➜  bin git:(support-make-failpoint) curl http://127.0.0.1:2379/pd/api/v1/fail/test
failed to GET: error on test: failpoint: failpoint does not exist
➜  bin git:(support-make-failpoint) curl -X PUT -d 'return(3474484975246189105)'  http://127.0.0.1:2379/pd/api/v1/fail/pd/server/exitCampaignLeader
➜  bin git:(support-make-failpoint) curl  http://127.0.0.1:2379/pd/api/v1/fail/                      
=return(3474484975246189105)
pd/server/exitCampaignLeader=return(3474484975246189105)


make pd-server
➜  bin git:(support-make-failpoint) tiup playground v7.1.1  --tiflash=0  --pd=1 --pd.binpath=./pd-server
➜  bin git:(support-make-failpoint) curl  http://127.0.0.1:2379/pd/api/v1/fail/
404 page not found
```

### Release note

```release-note
None.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -73,6 +73,11 @@ PD_SERVER_DEP += dashboard-ui
 pd-server: ${PD_SERVER_DEP}
 	CGO_ENABLED=$(BUILD_CGO_ENABLED) go build $(BUILD_FLAGS) -gcflags '$(GCFLAGS)' -ldflags '$(LDFLAGS)' -tags "$(BUILD_TAGS)" -o $(BUILD_BIN_PATH)/pd-server cmd/pd-server/main.go
 
+pd-server-failpoint:
+	@$(FAILPOINT_ENABLE)
+	$(MAKE) pd-server BUILD_TAGS="$(BUILD_TAGS) with_fail"
```
Comment (rleungx) [inline Makefile:78]: What about 
```
ifeq ($(FAILPOINT), 1)
	BUILD_TAGS += with_fail
endif
```
Then use `FAILPOINT=1 $(MAKE) pd-server`

Comment (nolouch) [reply Makefile:78]: It should make `failpoint-enale` first.

Comment (rleungx) [reply Makefile:78]: It doesn't conflict?

Comment (nolouch) [reply Makefile:78]: I think the current method is cleaner than the add some logic about if check and enable/disable failpoint rewrite in the `pd-server` command.

Comment (nolouch) [reply Makefile:78]: oh, I get your point.

**Review by HuSharp — APPROVED** [review]
LGTM!

**Review by rleungx — APPROVED** [review]

**Pull request merged**